src/ucp/rndv/rndv.c

```diff
@@ -1807,15 +1827,17 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
     int is_pipeline_rndv;
     uct_rkey_t uct_rkey;
 
-    UCP_WORKER_GET_REQUEST_BY_ID(&sreq, arg, rndv_rtr_hdr->sreq_id,
-                                 return UCS_OK, "RNDV RTR %p", rndv_rtr_hdr);
+    UCP_SEND_REQUEST_GET_BY_ID(&sreq, arg, rndv_rtr_hdr->sreq_id, 0,
+                               return UCS_OK, "RNDV RTR %p", rndv_rtr_hdr);
+
     ep        = sreq->send.ep;
     ep_config = ucp_ep_config(ep);
     put_zcopy = &ep_config->rndv.put_zcopy;
 
-    ucs_assertv(rndv_rtr_hdr->sreq_id == sreq->send.msg_proto.sreq_id,
-                "received local sreq_id 0x%"PRIx64" is not equal to expected sreq_id"
-                " 0x%"PRIx64, rndv_rtr_hdr->sreq_id, sreq->send.msg_proto.sreq_id);
+    ucs_assertv(rndv_rtr_hdr->sreq_id == ucp_send_request_get_local_id(sreq),
```

Comment:
can remove this assert, it's already checked in UCP_SEND_REQUEST_GET_BY_ID

Reply:
done